src/rgw/rgw_data_sync.cc

```diff
@@ -1014,7 +1014,7 @@ class RGWDataSyncSingleEntryCR : public RGWCoroutine {
         if (error_repo && !error_repo->append(raw_key)) {
           ldout(sync_env->store->ctx(), 0) << "ERROR: failed to log sync failure in error repo: retcode=" << retcode << dendl;
         }
-      } else if (error_repo && remove_from_repo) {
+      } else if (error_repo && remove_from_repo && (retcode >= 0)) {
```

Comment:
i think [this code in RGWBucketSyncSingleEntryCR](https://github.com/Jerry7X/ceph/blob/c43c144a245522e56d5c2c5882335c86f3c0f0ce/src/rgw/rgw_data_sync.cc#L2295-L2301) will prevent us from returning `ENOENT` from `RGWBucketShardIncrementalSyncCR`

it's the `ENOENT` errors from `RGWGetBucketInstanceInfoCR` in `RGWRunBucketSyncCoroutine` that we're worried about here - if we can't find the bucket instance (locally or on meta master zone), then we shouldn't keep trying to sync it